avaframe/ana3AIMEC/aimecTools.py

```diff
@@ -1059,3 +1064,200 @@ def getMaxMeanValues(rasterdataA, rasterArea):
+    rasterTransfo['l'] = rasterTransfo['l']*cellSizeSL
+    rasterTransfo['gridx'] = rasterTransfo['gridx']*cellSizeSL
+    rasterTransfo['gridy'] = rasterTransfo['gridy']*cellSizeSL
+    rasterTransfo['rasterArea'] = rasterTransfo['rasterArea']*cellSizeSL*cellSizeSL
+    # (x,y) coordinates of the resampled avapth (centerline where l = 0)
+    n = np.shape(rasterTransfo['l'])[0]
+    indCenter = int(np.floor(n/2))
+    rasterTransfo['x'] = rasterTransfo['gridx'][:, indCenter]
+    rasterTransfo['y'] = rasterTransfo['gridy'][:, indCenter]
+
+    return rasterTransfo
+
+
+def findStartOfRunoutArea(dem, rasterTransfo, cfgSetup, splitPoint):
+    """ find start of runout area point using splitPoint
+        add info on x, y coordinates of point, angle, index
+
+        Parameters
+        -----------
+        dem: dict
+            dictionary with DEM header and data
+        rasterTransfo: dict
+            domain transformation info
+        cfgSetup: configparser object
+            configuration for aimec
+        splitPoint: dict
+            dictionary with split Point coordinates
+
+        Returns
+        --------
+        rasterTransfo: dict
+            updated dictionary with info on start of runout location,...
+
+    """
+
+    # fetch input parameters from config
+    startOfRunoutAreaAngle = cfgSetup.getfloat('startOfRunoutAreaAngle')
+
+    # add 'z' coordinate to the centerline
+    rasterTransfo, _ = geoTrans.projectOnRaster(dem, rasterTransfo)
+
+    # find projection of split point on the centerline
+    projPoint = geoTrans.findSplitPoint(rasterTransfo, splitPoint)
+    rasterTransfo['indSplit'] = projPoint['indSplit']
+
+    # prepare find start of runout area points
+    angle, tmp, ds = geoTrans.prepareAngleProfile(startOfRunoutAreaAngle, rasterTransfo)
+
+    # find the runout point: first point under startOfRunoutAreaAngle
+    indStartOfRunout = geoTrans.findAngleProfile(tmp, ds, cfgSetup.getfloat('dsMin'))
+
+    # add info to rasterTransfo dict
+    rasterTransfo['indStartOfRunout'] = indStartOfRunout
+    rasterTransfo['xBetaPoint'] = rasterTransfo['x'][indStartOfRunout]
+    rasterTransfo['yBetaPoint'] = rasterTransfo['y'][indStartOfRunout]
+    rasterTransfo['startOfRunoutAreaAngle'] = angle[indStartOfRunout]
+
+    log.info('Start of run-out area at the %.2f ° point of coordinates (%.2f, %.2f)' %
+             (rasterTransfo['startOfRunoutAreaAngle'], rasterTransfo['xBetaPoint'], rasterTransfo['yBetaPoint']))
+
+    return rasterTransfo
+
+def makeDomainTransfoOnTheGo(avaDir, dem, cfgSetup, pathDict):
+    """ Make domain transformation
+
+    This function returns the information about the domain transformation
+    Data given on a regular grid is projected on a nonuniform grid following
+    a polyline to end up with "straightend raster"
+
+    Parameters
+    ----------
+    avaDir: pathlib path or str
+        path to avalanche directory
+    dem: dict
+        dictionary with header and raster data
```

Comment:
This is almost the same as makeDomainTrans.
Could we try to only have one function. I think it should be possible. 
If not (but I would prefer to avoid it) can you use the new sub functions (setAvaPath...) in makeDomainTrans

Reply:
made an issue here